### PR TITLE
"Site" parameter is optional now: when not provided the data is sent to all connected Jiras

### PR DESCRIPTION
**Pull request by bgvozdev**
This is the first PR from series to make the configuration of the plugin simpler to the customers 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java [hunk 1]
```diff
@@ -78,22 +80,57 @@ public JiraDeploymentInfoSenderImpl(
...
 
     @Override
-    public JiraSendInfoResponse sendDeploymentInfo(final JiraDeploymentInfoRequest request) {
-        final String jiraSite = request.getSite();
+    public List<JiraSendInfoResponse> sendDeploymentInfo(final JiraDeploymentInfoRequest request) {
+        final List<JiraSendInfoResponse> responses = new LinkedList<>();
+        if (request.getSite() == null) {
+            List<String> jiraSites = siteConfigRetriever.getAllJiraSites();
+            if (jiraSites.size() >= 2 && request.getEnableGating()) {
+                responses.add(JiraDeploymentInfoResponse.failureGatingManyJiras());
```
[the remark this answers is not in the document]
Comment (thombergs) [reply src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java:88]: Feels like we should document this special case for the users somewhere (i.e. "if you want to use gating, then you need to specify the site in the build step)

**Review by thombergs — APPROVED** [review]
Thanks for the PR, it gave me an opportunity to get into the topic. I've raised some (probably dumb) questions. Feel free to merge, but I'd like to walk this PR through with you on Zoom some time, even if it's merged already.

This PR doesn't solve the issue to automatically send builds and deployments without specifiying a build step, yet, does it?

File: src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java [hunk 2]
```diff
@@ -270,9 +309,9 @@ private String getDeploymentState(final WorkflowRun build, @Nullable final Strin
         return associations;
     }
 
-    private List<Command> buildCommands(final Boolean enableGating) {
+    private List<Command> buildCommands(@Nullable final Boolean enableGating) {
         final ImmutableList.Builder<Command> commandsBuilder = ImmutableList.builder();
-        if (enableGating) {
+        if (enableGating != null && enableGating) {
             commandsBuilder.add(new Command("initiate_deployment_gating"));
```
[the remark this answers is not in the document]
Comment (bgvozdev) [reply src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java:315]: Correct @thombergs . It is described pretty well here:

https://support.atlassian.com/jira-service-management-cloud/docs/use-deployment-gating-with-jenkins/

File: src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bgvozdev) [reply src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java:88]: > Why are we failing this request with more than one Jira when gating is enabled?

This was the outcome of my talks with JSM:  https://atlassian.slack.com/archives/CMW12BVC5/p1641515342000900?thread_ts=1641338013.029100&cid=CMW12BVC5

> The deployment has happened at this point already, hasn't it?

No. At this point the deployment hasn't started yet and it is created by this request in "pending" state, please check https://support.atlassian.com/jira-service-management-cloud/docs/use-deployment-gating-with-jenkins/  and where `enableGating` is sent to DataDepot . Once it is approved in JSM, the gating-status call will allow Pipeline to proceed and eventually the deployment will be updated to either "Successful" or "Failed".

> We're just sending the data from the deployment to the Jira API, aren't we?

Correct. Data Depot will send it further on to StreamHub where JSM is listening to these events.

> Feels like we should document this special case for the users somewhere

Yeah, this is still on my plate to figure out how we are updating the docs.

[the remark this answers is not in the document]
Comment (bgvozdev) [reply src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java:88]: I've updated docs @thombergs , JFYI

**Review by thombergs — APPROVED** [review]

**Pull request merged**